winter/argument_resolver.py

```diff
@@ -21,6 +22,10 @@ def __init__(self, argument: ComponentMethodArgument):
 class ArgumentResolver(abc.ABC):
     """IArgumentResolver is used to map http request contents to controller method arguments."""
 
+    def __init__(self):
+        self._cache: typing.Dict[ComponentMethodArgument, typing.Any] = {}
```

Comment:
**Just a notice**: the field is not a part of the mandatory interface but it's injected to all inherited classes, so they are coupling more then it's necessary. If you want to reuse the cache for some reason (however it's just a dict) then it's a good idea to create a separate class (i.e. `ComponentMethodArgumentCache`) which will serve the particular purpose.